nautobot/core/graphql/schema.py

```diff
@@ -159,6 +167,34 @@ def extend_schema_type_null_field_choice(schema_type, model):
     return schema_type
 
 
+def extend_schema_type_filter(schema_type, model):
+    """Extend schema_type object to be able to filter on multiple levels of a query
+
+    Args:
+        schema_type (DjangoObjectType): GraphQL Object type for a given model
+        model (Model): Django model
+
+    Returns:
+        schema_type (DjangoObjectType)
+    """
+    for field_name in dir(model):
+        attr = getattr(model, field_name)
+        # Check attribute is a ManyToOne field
+        if not isinstance(attr, ReverseManyToOneDescriptor):
+            continue
+
+        child_schema_type = registry["graphql_types"].get(attr.field.model._meta.label_lower)
+        if child_schema_type:
+            resolver_name = f"resolve_{field_name}"
```

Comment:
How do things work if a field is nullable (and so a resolver is added by `extend_schema_type_null_field_choice`) and *also* filterable? Does one resolver clobber the other, or do both somehow coexist?

Reply:
This function will only touch the fields generated by related objects so as far as I understand these can't be nullable